lib/ClientConnection.h

```diff
@@ -146,13 +146,9 @@ class PULSAR_PUBLIC ClientConnection : public std::enable_shared_from_this<Clien
      * Close the connection.
      *
      * @param result all pending futures will complete with this result
-     * @param detach remove it from the pool if it's true
-     *
-     * `detach` should only be false when:
-     * 1. Before the connection is put into the pool, i.e. during the construction.
-     * 2. When the connection pool is closed
+     * @param constructCompleted whether the construction is completed
      */
-    void close(Result result = ResultConnectError, bool detach = true);
+    void close(Result result = ResultConnectError, bool constructCompleted = true);
```

Comment:
It's not safe to directly replace the `detach` parameter. There are many other places where the close method is still called with the wrong parameter meaning.
For example at `ClientConnection.cc` line 214: `close(ResultAuthenticationError, false);` is called but with the constructCompleted false, which should be true because it's in the constructor.

Reply:
I will fix them soon.

Reply:
> For example at ClientConnection.cc line 214: close(ResultAuthenticationError, false); is called but with the constructCompleted false, which should be true because it's in the constructor.

`constructCompleted` means if the construction completes. If it's in the constructor, it should be false because the construction does not complete. That's the trick I made to avoid many changes.

Reply:
Sorry, it seems that I have given the wrong example. We need to take care of this: https://github.com/apache/pulsar-client-cpp/blob/6d47e9495c071975d94fc284d6bfb6bbe9c7501a/lib/ConnectionPool.cc#L57

This will incorrectly cause the connection closure to be skipped.

Reply:
Good catch! I think we should add a separated method instead of an overload.